furnace-service-provider/src/main/java/org/jboss/windup/web/furnaceserviceprovider/FurnaceExtension.java

```diff
@@ -0,0 +1,392 @@
+                    {
+                        return it.getInjectionPoints();
+                    }
+
+                    @Override
+                    public boolean isNullable()
+                    {
+                        return false;
+                    }
+
+                    @Override
+                    public Set<Type> getTypes()
+                    {
+                        return Collections.singleton(type);
+                    }
+
+                    @Override
+                    public Set<Annotation> getQualifiers()
+                    {
+                        Set<Annotation> qualifiers = new HashSet<>();
+                        qualifiers.add(new AnnotationLiteral<FromFurnace>(){});
+                        return qualifiers;
+                    }
+
+                    @Override
+                    public Class<? extends Annotation> getScope()
+                    {
+                        return ApplicationScoped.class;
+                    }
+
+                    @Override
+                    public String getName()
+                    {
+                        String name = type.getSimpleName();
+                        return name.substring(0, 1).toLowerCase() + name.substring(1);
+                    }
+
+                    @Override
+                    public Set<Class<? extends Annotation>> getStereotypes()
+                    {
+                        return Collections.emptySet();
+                    }
+
+                    @Override
+                    public boolean isAlternative()
+                    {
+                        return false;
+                    }
+
+                    @Override
+                    public T create(CreationalContext<T> creationalContext)
+                    {
+                        T instance = it.produce(creationalContext);
+                        it.inject(instance, creationalContext);
+                        it.postConstruct(instance);
+                        return instance;
+                    }
+
+                    @Override
+                    public void destroy(T instance, CreationalContext<T> creationalContext)
+                    {
+                        it.preDestroy(instance);
+                        it.dispose(instance);
+                        creationalContext.release();
+                    }
+
+                    @Override
+                    public String toString()
+                    {
+                        return "Bean: type = " + type.getCanonicalName();
+                    }
+                });
+            }
+            catch (Throwable t)
+            {
+                t.printStackTrace();
+            }
+        }
+
+        if (type.getInterfaces() != null)
+        {
+            for (Class<?> iface : type.getInterfaces())
+            {
+                addService(duplicateCheck, abd, beanManager, iface);
+            }
+        }
```

Comment:
This registers the interfaces tree recursively, right? The superclass is not necessary?